### ISSUE TITLE
L2 weight decay for Q

### ISSUE THREAD

**Issue by Anjum48** [pemami4911/deep-rl#9]
The paper mentions "For Q we included L2 weight decay of 0.01 and used a discount factor of gamma = 0.99"

Does that mean that we need to add L2 regularisation to each layer in the critic network?

Maybe something like this in `create_critic_network`
```
net = tflearn.add_weights_regularizer(net, 'L2', weight_decay=0.01)
```

**Comment by pemami4911** [comment]
It's not necessary for the simple pendulum example, but for more complex tasks you should add that.

**Comment by Anjum48** [comment]
Cool - thanks! I think I've got it to work now for my example (non-pendulum). For your code just add this at line 177 in ddpg.py

```
self.L2 = np.sum([tf.nn.l2_loss(v) for v in self.network_params if "W" in v.name])
self.loss = tflearn.mean_square(self.predicted_q_value, self.out) + L2_DECAY * self.L2
```

**Comment by AlEmerich** [comment]
I have the same question in my mind @Anjum48, and I didn't find the anwser in your public repo. 
Did you add L2 regularizer in every layer or just the ouput ? And what about the actor network ?

**Comment by Anjum48** [comment]
Hi @AlEmerich - in my repo I actually skipped the L2 regularisation for DDPG since it seemed to work fine without it, however if you look at my PPO example I've added it there to every `dense` layer in the actor and critic using `kernel_regularizer=tf.contrib.layers.l2_regularizer(L2_REG)`

**Comment by AlEmerich** [comment]
Hi (Sorry for my bad manners :) ) and thank you for the precision !
